pkg/isbsvc/jetstream_service.go

```diff
@@ -120,11 +120,23 @@ func (jss *jetStreamSvc) CreateBuffersAndBuckets(ctx context.Context, buffers, b
 			if !errors.Is(err, nats.ErrStreamNotFound) {
 				return fmt.Errorf("failed to query information of stream %q during buffer creating, %w", streamName, err)
 			}
+			// get the retention policy from the stream config
+			retention := nats.RetentionPolicy(v.GetInt("stream.retention"))
+			// default discard policy is set to DiscardOld
+			discard := nats.DiscardOld
+
+			// Based on the retention policy we use the following discard policy
+			// 1) Limits/Interest Policy -> DiscardOld
+			// 2) WorkQueuePolicy -> DiscardNew
+			if retention == nats.WorkQueuePolicy {
+				discard = nats.DiscardNew
+			}
```

Comment:
what about Interest? Why should it be `DiscardOld`? please document

Reply:
I went ahead as it was default, but I realised it didn't seem to be correct.
We should use it with DiscardNew instead
I have added few lines to document, please take a look @vigith